client-js/app/elements/labrad-plot.ts

```diff
@@ -154,49 +240,129 @@ export class Plot extends polymer.Base {
     for (let row of data) {
       this.splice('data', this.data.length, 0, row);
     }
-    this.plotData(data, lastData);
+    this.plotData(data);
+
+    if (!this.haveZoomed) {
+      this.resetZoom();
+    } else {
+      this.projectGraphPositions();
+    }
   }
 
 
-  private createPlot(
-      area: HTMLElement, totWidth: number, totHeight: number): void {
+  /**
+   * Redraw the plot and attach resize to the window resize event.
+   * Fires when the component is attached to the DOM.
+   */
+  attached() {
+    this.render();
+    window.addEventListener('resize', (e) => this.resizePlot());
+  }
+
+
+  /**
+   * Resets the current zoom level to fit the data.
+   */
+  resetZoomControl() {
```

Comment:
You mean attached()? or resetZoomControl()? 

attached() is called when the component loads.
resetZoomControl() is called from the html UI element.